src/inverted-index.js

```diff
@@ -0,0 +1,116 @@
+class InvertedIndex{
+	
+	constructor() {
+		/**
+		 * Initialize an empty array to hold the
+		 * concantenated words formed from title and text
+		 * contents.
+		 */ 
+		this.indexWords = [];
+		/**
+		 * Initialize an empty object to hold the mapping
+		 * of the words to their file location 
+		 */
+		this.index = {};
+		/**
+		 * Initialize an empty array to hold the 
+		 * search results for words
+		 */
+		this.searchResults = [];
+	}
+	
+	/**
+	 * This Method returns an array of tokens/words from the provided input
+	 * after removing special characters and white spaces
+	 * @params{String} str - String to be tokonized
+	 */
+	tokenize(str) {
+	    var cleanString = str.trim().replace(/-/g, " ").replace(/[.,\/#!$%\^&@\*;:'{}=\_`~()]/g, "").toLowerCase().split(" ").sort();
+	    return cleanString;
+	}
+	
+	/**
+	 * This method returns an array of tokenized unique words
+	 * of the provided input 
+	 * @params{String} str - The string to be filtered
+	 */
+	uniqueWords(str) {
+		var tokens = this.tokenize(str);
+		return tokens.filter(function (item, pos) {
+      		return tokens.indexOf(item) === pos;
+		});
+	}
+
+	/**
+	 * This method creates an index and updates the index object
+	 * Also populates the indexWords array with contents of supplied data
+	 * @params{Array} jsonData - The contents of the JSON file that is to be indexed
+	 */
+	createIndex(jsonData) {
+		for (let i of jsonData) {
+			if(i.text) {
+				//Iterate through the jsonData and store contents in an array
+				this.indexWords.push(i.title.toLowerCase() + " " + i.text.toLowerCase());
```

Comment:
not necessarily, **_this.indexWords_** is used in **_createIndex_**

_self.indexWords.forEach(function(document, pos)....  _

_document.indexOf(word)_ will return -1 for comparison like
['Pol'].indexOf('pol')